packages/theme/src/client/modules/info/components/TOC.ts

```diff
@@ -134,13 +135,39 @@ export default defineComponent({
       );
     });
 
+    const updateTocMarker = (): void => {
+      if (!toc.value) return;
+      const tocTop = toc.value.getBoundingClientRect().top;
+      const tocScrollTop = toc.value.scrollTop;
+
+      const activeTocItem = document.querySelector(".toc-item.active");
+
+      if (!activeTocItem) return;
+      const activeTocItemTop = activeTocItem.getBoundingClientRect().top;
+
+      const tocMarker = document.querySelector(".toc-marker") as HTMLElement;
+
+      tocMarker.style.top = `${activeTocItemTop - tocTop + tocScrollTop}px`;
+    };
+
+    onUpdated(() => {
+      updateTocMarker();
+    });
```

Comment:
So why using `onUpdated`? Is there any reason to do this instead  of manually create a `watch` or `watchEffect` with certain deps?

Reply:
There is no particular reason. Currently I do not fully understand the reactivity of this component. Would you be kind enough to refine it?